lib/datasets/estatjp.rb

```diff
@@ -0,0 +1,302 @@
+# frozen_string_literal: true
+
+require 'digest/md5'
+require 'net/http'
+require 'uri'
+require 'json'
+
+module Datasets
+  Record = Struct.new(:id, :name, :values)
+
+  # Estat module
+  module Estatjp
+    # configuration injection
+    module Configuration
+      attr_accessor :app_id
+
+      #
+      # configuration for e-Stat API
+      # See detail at https://www.e-stat.go.jp/api/api-dev/how_to_use (Japanese only).
+      # @example
+      #  Datasets::Estat.configure do |config|
+      #   # put your App ID for e-Stat app_id
+      #   config.app_id = 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+      #  end
+      #
+      def configure
+        yield self
+      end
+    end
+
+    extend Configuration
+
+    # wrapper class for e-Stat API service
+    class JSONAPI < Dataset
```

Comment:
I think that "JSON API" is an implementation detail.
Can we hide this?

How about the following name?

  * `StatsData` because API endpoint URL is `.../getStatsData`

We will be able to add the following datasets later:

  * `StatsList` for `.../getStatsList`
  * `MetaInfot` for `.../getMetaInfo`
  * `DataCatalog` for `.../getDataCatalog`
  * `LDO` (or `LocalOpenData`) for https://data.e-stat.go.jp/lodw/

Reply:
Yes, it's worth changing for scalability of supporting other endpoints.